Composer/packages/client/src/components/CreationFlow/TemplateDetailView.tsx

```diff
@@ -58,6 +64,31 @@ type TemplateDetailViewProps = {
 const templateDocUrl = 'https://aka.ms/localComposerTemplateDoc';
 
 export const TemplateDetailView: React.FC<TemplateDetailViewProps> = (props) => {
+  const { setSelectedTemplateVersion } = useRecoilValue(dispatcherState);
+  const selectedTemplateVersion = useRecoilValue(selectedTemplateVersionState);
+
+  useEffect(() => {
+    props.template?.package?.packageVersion && setSelectedTemplateVersion(props.template.package.packageVersion);
+  }, [props.template]);
+
+  const renderVersionButton = () => {
+    const availableVersions = props.template?.package?.availableVersions || ([] as string[]);
+    availableVersions;
+    const versionOptions = {
```

Comment:
Do you feel there would be value in memoizing this with a dependancy on props.template? The render for versionButton does not need change for every time the component rerenders right?

Reply:
Yeah that makes sense, I have made this change